src/promptflow-devkit/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -70,10 +66,10 @@ def trace_collector(
             for scope_span in resource_span.scope_spans:
                 for span in scope_span.spans:
                     # TODO: persist with batch
-                    span = Span._from_protobuf_object(span, resource=resource)
+                    span = Span._from_protobuf_object(span, resource=resource, logger=logger)
+                    all_spans.append(copy.deepcopy(span))
```

Comment:
Here do the deepcopy to avoid the value change in "span._persist"? But if cloud_trace_only=true, maybe we don't need to do the copy? 
But I am not sure the impact of deepcopy, maybe is OK.

Reply:
got your point, I will add a if here